File: 5.implementation/redsun/src/main/webapp/static/js/services/PaymentModesService.js


/**
 * Controller for ${entity.name}
 **/


app.factory('paymentModeService', function($http, $rootScope) {
	return {
	
		// Insert.
		add: function(paymentMode) {
			var urlServer = $rootScope.makePostURL("/paymentmode/insert");
			return $http.post(urlServer, paymentMode);
		}
		
		// Update.
		, update: function(paymentModeId, paymentMode) {
			var urlServer = $rootScope.makePostURL("/paymentmode/update/" + paymentModeId);
			return $http.put(urlServer, paymentMode);
		}
		
		// Delete.
		, delete: function(paymentModeId) {
			var urlServer = $rootScope.makePostURL("/paymentmode/delete/" + paymentModeId);
			return $http.delete(urlServer);
		}
		
		// Get by Id.
		, getById: function(paymentModeId) {
			var url = $rootScope.makeGetURL("/paymentmode/getbyid/" + paymentModeId);
			return $http.get(url);
		}
		
		// List for page and filter.
		, getsForPageAndFilter: function(itemsPerPage, pageNo, paymentMode) {
			var url = $rootScope.makeGetURL("/paymentmode/list/page/filter/" + itemsPerPage + "/" + pageNo);
			return $http.get(url, { params: paymentMode });
		}
		
	}// return.
});

File: 5.implementation/VT/redsun-service/src/main/java/com/redsun/service/service/PaymentModesService.java
package com.redsun.service.service;

import com.redsun.service.entities.PaymentModes;
import com.redsun.service.entities.Result;

/**
 * PaymentModes Service interface
 */
public interface PaymentModesService {
	
    // Insert.
	Result insert(final PaymentModes paymentModes) ;

    // Update.
	Result update(final PaymentModes paymentModes) ;

    // Delete.
	Result delete(final Integer id) ;

	// Get By Id.
	Result getById(final Integer id) ;

    // Exists.
	Result exists(final Integer id) ;

    // Count.
	Result count() ;

	// List for page and filter.
	Result listPaymentModesForPageAndFilter(final int itemsPerPage, final int pageNo, final PaymentModes paymentModes);

}
